src/Notifier/BoolQueryDispatcher.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/*
+
+This file was created by developers working at BitBag
+
+Do you need more information about us and what we do? Visit our   website!
+
+We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+declare(strict_types=1);
+
+namespace BitBag\SyliusElasticsearchPlugin\Notifier;
+
+use BitBag\SyliusElasticsearchPlugin\EventListener\QueryCreatedEventInterface;
+use BitBag\SyliusElasticsearchPlugin\Factory\QueryCreatedEventFactoryInterface;
+use Elastica\Query\AbstractQuery;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+final class BoolQueryDispatcher implements BoolQueryDispatcherInterface
```

Comment:
Why `BoolQuery` as a prefix of name? Isn't the name too specific?

Reply:
We want dispatch exactly elastic `BoolQuery` I guess

Reply:
Is this possible to have it as a generic? To cover all queries dispatching event.